core/src/main/java/tech/pegasys/eth2signer/core/multikey/DirectoryBackedArtifactSignerProvider.java

```diff
@@ -143,26 +141,25 @@ private void cacheSigner(final String identifier) {
   }
 
   private <T> Collection<T> findSigners(
-      final DirectoryStream.Filter<? super Path> filter, final Function<Path, T> mapper) {
+      final Predicate<? super Path> filter, final Function<Path, T> mapper) {
     final Collection<T> signers = new HashSet<>();
 
-    try (final DirectoryStream<Path> directoryStream =
-        Files.newDirectoryStream(configsDirectory, filter)) {
-      for (final Path file : directoryStream) {
+    try (final Stream<Path> fileStream = Files.list(configsDirectory)) {
+      fileStream.filter(filter).parallel().forEach(signerConfigFile -> {
         try {
-          signers.add(mapper.apply(file));
+          LOG.debug("Loading {}", signerConfigFile);
+          signers.add(mapper.apply(signerConfigFile));
         } catch (final Exception e) {
-          renderException(e, file.getFileName().toString());
+          renderException(e, signerConfigFile.getFileName().toString());
         }
-      }
-      return signers;
-    } catch (final IOException | SecurityException e) {
-      LOG.warn("Error searching for signing metadata files: {}", e.getMessage());
-      return Collections.emptySet();
+      });
+    } catch (final IOException e) {
+      LOG.error("Didn't want this one", e);
```

Comment:
Why has the error handling changed here? Don't we still want to return an empty set?

Reply:
Will fall through into the emptySet returns (now)